webhooks-extension/src/components/WebhookCreate/WebhookCreate.js

```diff
@@ -0,0 +1,698 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+
+import { Button, TextInput, Dropdown, Form, Tooltip, DropdownSkeleton, Modal, InlineNotification } from 'carbon-components-react';
+import { getNamespaces, getPipelines, getSecrets, getServiceAccounts, createWebhook, createSecret, deleteSecret } from '../../api';
+
+import AddAlt20 from '@carbon/icons-react/es/add--alt/20';
+import SubtractAlt20 from '@carbon/icons-react/es/subtract--alt/20';
+import ViewFilled from '@carbon/icons-react/es/view--filled/20';
+import ViewOffFilled from '@carbon/icons-react/es/view--off--filled/20';
+
+
+import './WebhookCreate.scss';
+
+class WebhookCreatePage extends Component {
+
+  constructor(props) {
+    super(props);
+    // turn off webhook created message
+    this.props.setShowNotification(false);
+
+    this.state = {
+      // variables to stop re-attempts to load
+      namespaceFail: false,
+      pipelineFail: false,
+      secretsFail: false,
+      serviceAccountsFail: false,
+      // selected values stored as these
+      name: '',
+      repository: '',
+      namespace: '',
+      pipeline: '',
+      gitsecret: '',
+      serviceAccount: '',
+      dockerRegistry: '',
+      // fetched data from api calls
+      apiNamespaces: '',
+      apiPipelines: '',
+      apiSecrets: '',
+      apiServiceAccounts: '',
+      // whether or not to show secret modals
+      showDeleteDialog: false,
+      showCreateDialog: false,
+      // whether or not to show error for delete with no secret selected
+      showNotification: false,
+      // error messages
+      notificationErrorMessage: "",
+      notificationStatus: 'success', // or error, or warning,
+      notificationStatusMsgShort: 'Secret deleted successfully',
+      // create secret vars
+      newSecretName: '',
+      newTokenValue: '',
+      createSecretDisabled: true,
+      // toggle access token 'password' or 'text'
+      pwType: 'password',
+      visibleCSS: 'token-visible',
+      invisibleCSS: 'token-invisible'
+    };
+  }
+
+  async fetchNamespaces() {
```

Comment:
Can these fetch/get methods be extracted into a different file to be used by the component? I think that could make the code easier to read and would reduce the amount of code in this one file, though I do see they are altering state so might not be an option currently

Reply:
I'd rather any refactor was done later ... we likely need to look at whether we could/should introduce any redux usage which might further alter things